### PR TITLE
Configure esp32 format and add component build

### PR DESCRIPTION
**Pull request by q349wang**
Sets up the esp32 format for adding components to the build and adds some of the hal base libs

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,14 +1,24 @@
 cmake_minimum_required(VERSION 3.16)
 
+# API Key for SFMTC api.511.org
+# should be in antioch build dir
+
+# Temporary no key
+file(WRITE "${CMAKE_CURRENT_SOURCE_DIR}/build/api_key_511" "FOOBAR")
```
Comment (Victor-C-Zhang) [inline CMakeLists.txt]: don't commit this one

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,14 +1,24 @@
...
-set(EXTRA_COMPONENT_DIRS "${CMAKE_CURRENT_SOURCE_DIR}/espgazer" "${CMAKE_CURRENT_SOURCE_DIR}/fw" "${CMAKE_CURRENT_SOURCE_DIR}/gfx")
+set(EXTRA_COMPONENT_DIRS
+  "${CMAKE_CURRENT_SOURCE_DIR}/espgazer"
+  "${CMAKE_CURRENT_SOURCE_DIR}/fw"
+  "${CMAKE_CURRENT_SOURCE_DIR}/gfx"
+  "${CMAKE_CURRENT_SOURCE_DIR}/esp_components")
 
 include($ENV{IDF_PATH}/tools/cmake/project.cmake)
 
-project(antioch)
```
Comment (Victor-C-Zhang) [inline CMakeLists.txt:9]: don't we need this on regular computer build?

File: esp_components/input-hal/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+FILE(GLOB SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../../device/esp32/input/*.cpp")
```
Comment (Victor-C-Zhang) [inline esp_components/input-hal/CMakeLists.txt:1]: are you able to do `${PROJECT_SOURCE_DIR}` insstead of `CURRENT_SOURCE_DIR/../..`

File: esp_components/libwifi/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,6 @@
+FILE(GLOB SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../../connectivity/wifi/src/*.cpp")
```
Comment (Victor-C-Zhang) [inline esp_components/libwifi/CMakeLists.txt:1]: same

File: esp_components/wifi-hal/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,4 @@
+FILE(GLOB SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../../device/esp32/wifi/*.cpp")
```
Comment (Victor-C-Zhang) [inline esp_components/wifi-hal/CMakeLists.txt:1]: same

File: sdkconfig [hunk 6]
```diff
@@ -480,7 +480,8 @@ CONFIG_COMPILER_FLOAT_LIB_FROM_GCCLIB=y
 CONFIG_COMPILER_OPTIMIZATION_ASSERTION_LEVEL=2
 # CONFIG_COMPILER_OPTIMIZATION_CHECKS_SILENT is not set
 CONFIG_COMPILER_HIDE_PATHS_MACROS=y
-# CONFIG_COMPILER_CXX_EXCEPTIONS is not set
+CONFIG_COMPILER_CXX_EXCEPTIONS=y
+CONFIG_COMPILER_CXX_EXCEPTIONS_EMG_POOL_SIZE=0
```
Comment (Victor-C-Zhang) [inline sdkconfig:484]: what is emg pool

**Review by Victor-C-Zhang — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 2 above)
Comment (q349wang) [reply CMakeLists.txt:9]: Its still there, line 26

File: CMakeLists.txt [hunk 7]
```diff
@@ -1,12 +1,13 @@
...
-# should be in antioch build dir
+# should be in antioch project dir
 
-# Temporary no key
-file(WRITE "${CMAKE_CURRENT_SOURCE_DIR}/build/api_key_511" "FOOBAR")
-
-file(STRINGS "${CMAKE_CURRENT_SOURCE_DIR}/build/api_key_511" API_KEY_511)
+if (EXISTS "${PROJECT_SOURCE_DIR}/api_key_511")
+file(STRINGS "${PROJECT_SOURCE_DIR}/api_key_511" API_KEY_511)
+else()
```
Comment (Victor-C-Zhang) [inline CMakeLists.txt:8]: bro just get a key

**Review by Victor-C-Zhang — APPROVED** [review]

**Pull request merged**